### ISSUE TITLE
Encode special characters in URLs

### ISSUE THREAD

**Issue by rottame** [rails/rails#55436]
Pinning a file with a '+' character in its name results in an invalid URL.

```
pin 'foo', to: 'packages/foo+bar.js'
```

![image](https://github.com/user-attachments/assets/1b38cbfc-ecf8-4361-a27f-9fdf2e827b9b)

The resulting URL leads to an HTTP 404 response. 

The correct URL should contain the '+' character encoded as '%2B'.

![image](https://github.com/user-attachments/assets/e1cacde0-255d-42c1-8c38-1f6c7dac7bd4)

**Comment by rafaelfranca** [comment]
This isn't a bug in importmap-rails. Rails assets helpers don't encode the URL passed to it:

```irb
>> helper.javascript_include_tag("http://foo.com/omg+bar.js")
=> "<script src=\"http://foo.com/omg+bar.js\"></script>"
>> helper.asset_path("http://foo.com/omg+bar.js")
=> "http://foo.com/omg+bar.js"
```

This is a bug in Rails and its two assets pipeline.

I'll move this issue there.

**Comment by adrienpoly** [comment]
I gave it a try in #56053

**Comment by matthewd** [comment]
I don't think this is a valid issue. A URL path can contain `+`; if that's the path we're given, it seems like that's the path we should use?

**Comment by zzak** [comment]
👍 I think suddenly escaping URLs that were unescaped before is a breaking change
